lib/modules/smart/shared_launch_tests.rb

```diff
@@ -76,8 +76,13 @@ def validate_token_response_contents(token_response, require_expires_in:)
           assert missing_scopes.empty?, "Token exchange response did not include expected scopes: #{missing_scopes}"
         end
 
-        extra_scopes = actual_scopes - expected_scopes
-        assert extra_scopes.empty?, "Token response contained unrequested scopes: #{extra_scopes.join(', ')}"
+        # During a token refresh scopes provided must be a strict sub-set of the scopes granted in the original launch.
+        # This does not apply to the original token exchange
+        # See: https://github.com/onc-healthit/inferno/issues/464 and other related issues tagged linked to that issue.
+        if check_unrequested_scopes
+          extra_scopes = actual_scopes - expected_scopes
```

Comment:
I think `expected_scopes` probably should be `@instance.received_scopes`, because `expected_scopes` here is what was requested in the original request I think, not what was actually granted (received).  So if you have a situation where in the initial authorization request, the app requests `patient/*.read`, the server grants `patient/Condition.read` and `patient/Patient.read`.  On the token refresh, I'd expect it to be a subset of `patient/Condition.read` and `patient/Patient.read`, not `patient/*.read`, no?

This is complicated because of how the code is shared though, so we need to be careful here.

Reply:
Good catch.  Looking into a fix now, but I'm wondering if we'll also have issues if we are granted something with a wildcard originally and then a subset later